src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
   // write code here;
+  const groupedObjects = {};
+
+  items.forEach((item) => {
+    if (!((item[key]) in groupedObjects)) {
+      groupedObjects[item[key]] = [];
+    }
+
+    groupedObjects[item[key]].push(item);
```

Comment:
Let's think in reverse, it looks better agree, doesn't it?
```suggestion
    if (item[key] in groupedObjects)) {
      groupedObjects[item[key]].push(item);
    }

    groupedObjects[item[key]] = [];
```

Reply:
Я переробив більш наглядно, але якщо я так міняю місцями то половина тестів не проходить. Вже по всякомі пробував